File: telegram-service/src/main/java/de/nak/home_assistant/commands/Hue.java
package de.nak.home_assistant.commands;

import com.pengrad.telegrambot.TelegramBot;
import com.pengrad.telegrambot.model.Update;
import com.pengrad.telegrambot.request.SendMessage;
import de.nak.home_assistant.actions.DeviceList;
import de.nak.home_assistant.models.database.ModuleResponse;
import de.nak.home_assistant.models.telegram.Command;
import de.nak.home_assistant.services.database.ModuleService;

import java.util.Optional;

public class Hue extends Command {

    private final TelegramBot bot;

    public Hue(TelegramBot bot) {
        this.bot = bot;
        this.name = "hue";
        this.description = "Öffnet das Hue Service Menü";
    }

    public Hue() {
        this(null);
    }

    public void reply(Update update) {
        long userId = update.message().from().id();
        long chatId = update.message().chat().id();

        Optional<ModuleResponse> moduleResponse = new ModuleService().findModuleByName(this.name);
        if (!moduleResponse.isPresent()) {
            SendMessage msg = new SendMessage(chatId, "Interner Fehler (es wurde kein Modul mit diesem Name gefunden)");
            bot.execute(msg);
            return;
        }

        int moduleId = moduleResponse.get().getId();
        SendMessage message = DeviceList.generateMessage(userId, chatId, moduleId);
        bot.execute(message);
    }
}

File: telegram-service/src/commands/hue.ts
import { Message } from "node-telegram-bot-api";
import { sendDeviceList } from "../actions/device-list";
import { bot } from "../bot";
import { i18n } from "../i18n";
import { findModuleByName } from "../services/module";
import { Command } from "../telegram/command";

export const Hue: Command = () => {
  const translations = i18n();

  return {
    command: translations.hue.name,
    description: translations.hue.description,
  };
};

export const replyToHue = async ({ from, chat }: Message) => {
  const translations = i18n();

  return findModuleByName(Hue.name)
    .then((module) => sendDeviceList(from.id, chat.id, module.id))
    .catch((e) => bot.sendMessage(chat.id, `${translations.internalError}`));
};
